gson/src/test/java/com/google/gson/JsonPrimitiveTest.java

```diff
@@ -316,4 +316,31 @@ public void testDeepCopy() {
     JsonPrimitive a = new JsonPrimitive("a");
     assertThat(a).isSameInstanceAs(a.deepCopy()); // Primitives are immutable!
   }
+
+  @Test
+  public void testBigDecimalEquals() {
+    JsonPrimitive small = new JsonPrimitive(1.0);
+    JsonPrimitive large = new JsonPrimitive(2.0);
+    assertThat(small.equals(large)).isFalse();
+
+    BigDecimal doubleMax = BigDecimal.valueOf(Double.MAX_VALUE);
+    JsonPrimitive smallBD = new JsonPrimitive(doubleMax.add(new BigDecimal("100.0")));
+    JsonPrimitive largeBD = new JsonPrimitive(doubleMax.add(new BigDecimal("200.0")));
+    assertThat(smallBD.equals(largeBD)).isFalse();
+  }
+
+  @Test
+  public void testBigDecimalEqualsZero() {
+    assertThat(new JsonPrimitive(new BigDecimal("0.0"))
+        .equals(new JsonPrimitive(new BigDecimal("0.00")))).isTrue();
+
+    assertThat(new JsonPrimitive(new BigDecimal("0.00"))
+        .equals(new JsonPrimitive(Double.valueOf("0.00")))).isTrue();
+  }
+
+  @Test
+  public void testEqualsDoubleNaNAndBigDecimal() {
+    assertThat(new JsonPrimitive(Double.NaN)
+            .equals(new JsonPrimitive(new BigDecimal("1.0")))).isFalse();
```

Comment:
Indentation here seems to be wrong:
```suggestion
        .equals(new JsonPrimitive(new BigDecimal("1.0")))).isFalse();
```